src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -191,6 +204,10 @@ class AuthScreens extends React.Component {
             return false;
         }
 
+        if (this.props.isCreatingWorkspace) {
```

Comment:
Good point. I put up this PR before I had figured out that Onyx.clear() was resetting the session silently. I will run the tests with #8611 merged into the feature branch instead of this PR, and if it works then we won't need these changes.